### ISSUE TITLE
Manifest is not a valid M3U8 file

### ISSUE THREAD

**Issue by difelice** [brunobasto/Peerflix.bundle#2]
Hi @brunobasto I finally got *peerflix* working, I changed the way the host is retrieved (https://github.com/brunobasto/Peerflix.bundle/compare/master...difelice:hostname-ip) and I had to mannually set the *peerflix* path (mine is under /opt/local/bin) for now.

Video stream now works via VLC but I got the following error on *Plex Web App*:

```
Manifest is not a valid M3U8 file
https://192-168-1-80.fd990b900d664782afcad71073da4351.plex.direct:32400/video/peerflix/PlayHLS.m3u8?magnet=OBFUSCATED_MAGNET_LINK&startedAt=1434078219&streamURL=http%3A%2F%2F192.168.1.80%3A8926%2F&indirect=1&mediaInfo=%7B%22audio_channels%22%3A%20null%2C%20%22protocol%22%3A%20null%2C%20%22optimized_for_streaming%22%3A%20null%2C%20%22video_frame_rate%22%3A%20null%2C%20%22duration%22%3A%20null%2C%20%22height%22%3A%20null%2C%20%22width%22%3A%20null%2C%20%22container%22%3A%20null%2C%20%22audio_codec%22%3A%20null%2C%20%22aspect_ratio%22%3A%20null%2C%20%22video_codec%22%3A%20null%2C%20%22video_resolution%22%3A%20null%2C%20%22bitrate%22%3A%20null%7D&X-Plex-Token=PhJ8F1bxN3eKYUdLJqyz
```
Note: I obfuscated the magnet link on purpose.

I noted that all the media attributes are **null**, but not sure that's the issue. It will be nice if you can compare it with a working one. I will keep testing to figure out what's wrong.

Thanks a lot.
Cheers

**Comment by brunobasto** [comment]
That's interesting. The error ``Manifest is not a valid M3U8 file`` doesn't come from my code, neither from peerflix's code (https://github.com/mafintosh/peerflix/). I did a quick search in the internet and found this may be a flash player error: http://forums.opera.com/discussion/1851793/flash-player-manifest-is-not-a-valid-m3u8-/p1

That makes me think you are not using the HTML5 player. Can you test using it? ![Experimental Player](http://ibin.co/24zzpD8DZS8x)

**Comment by brunobasto** [comment]
Also, don't forget to disable Direct Play and Direct Stream: ![Disable Direct Play and Direct Stream](http://ibin.co/25026xRemv3N)

**Comment by difelice** [comment]
Yes I disabled Direct Play and Direct Stream. I'll try changing the Experimental Player option. Thanks!

**Comment by brunobasto** [comment]
Try to test with different videos. I found that the trending ones tend to work better.

**Comment by brunobasto** [comment]
@difelice 

Can you send me a pull request with the fix you added for the way the host is retrieved?

Were you able to get it to work at all?

Thanks,
